wasmtime/_func.py

```diff
@@ -201,20 +209,22 @@ def invoke(idx, params_ptr, results_ptr, params):  # type: ignore
 
     try:
         for i in range(0, len(param_tys)):
-            params.append(extract_val(Val(params_ptr[i])))
+            params.append(Val(params_ptr[i]).value)
         results = func(*params)
         if len(result_tys) == 0:
             if results is not None:
                 raise WasmtimeError(
                     "callback produced results when it shouldn't")
         elif len(result_tys) == 1:
-            val = Val.__convert__(result_tys[0], results)
+            val = Val._convert(result_tys[0], results)
+            val._owner = "passing ownership into Wasm"
             results_ptr[0] = val._raw
         else:
             if len(results) != len(result_tys):
                 raise WasmtimeError("callback produced wrong number of results")
             for i, result in enumerate(results):
-                val = Val.__convert__(result_tys[i], result)
+                val = Val._convert(result_tys[i], result)
+                val._owner = "passing ownership into Wasm"
```

Comment:
Instead of setting the `_owner` here so it doesn't do anything in the destructor, could this instead perhaps do something like `take()` where it cancels the destructor internally? I think ideally only `_val.py` should be reaching into the internal fields of `Val`